core/server/worker/src/test/java/alluxio/worker/block/BlockMasterClientTest.java

```diff
@@ -77,4 +162,343 @@ public void convertBlockListMapToProtoMergeDirsInSameTier() {
+              BlockStoreLocationProto locationProto = entry.getKey();
+              BlockStoreLocation location = new BlockStoreLocation(
+                  locationProto.getTierAlias(), 0, locationProto.getMediumType());
+              List<Long> blockIdList = currentBlocksOnLocation.get(location);
+              assertTrue(blockIdList != null
+                  && blockIdList.containsAll(entry.getValue().getBlockIdList()));
+            }
+
+            assertEquals(lostStorage.entrySet().stream()
+                .collect(Collectors.toMap(Map.Entry::getKey,
+                    e -> StorageList.newBuilder().addAllStorage(e.getValue()).build())),
+                request.getLostStorageMap());
+
+            registeredWorkerIds.add(request.getWorkerId());
+            responseObserver.onNext(RegisterWorkerPResponse.newBuilder().build());
+            responseObserver.onCompleted();
+          }
+
+          @Override
+          public StreamObserver<RegisterWorkerPRequest> registerWorkerStream(
+              StreamObserver<RegisterWorkerPResponse> responseObserver) {
+            return new StreamObserver<RegisterWorkerPRequest>() {
+              @Override
+              public void onNext(RegisterWorkerPRequest value) {
+                registeredWorkerIds.add(value.getWorkerId());
+                responseObserver.onNext(RegisterWorkerPResponse.newBuilder().build());
+              }
+
+              @Override
+              public void onError(Throwable t) {
+              }
+
+              @Override
+              public void onCompleted() {
+                responseObserver.onCompleted();
+              }
+            };
+          }
+        });
+
+    BlockMasterClient client = new BlockMasterClient(
+        MasterClientContext.newBuilder(ClientContext.create(mConf)).build()
+    );
+
+    if (stream) {
+      client.registerWithStream(
+          workerId,
+          storageTierAliases,
+          totalBytesOnTiers,
+          usedBytesOnTiers,
+          currentBlocksOnLocation,
+          lostStorage,
+          configList
+      );
+    } else {
+      client.register(
+          workerId,
+          storageTierAliases,
+          totalBytesOnTiers,
+          usedBytesOnTiers,
+          currentBlocksOnLocation,
+          lostStorage,
+          configList);
+    }
+
+    assertEquals(1, registeredWorkerIds.size());
+    assertEquals(workerId, (long) registeredWorkerIds.get(0));
+  }
+
+  @After
+  public void cleanUp() {
+    if (mServer != null && mServer.isServing()) {
+      mServer.shutdown();
+    }
+
+    mServer = null;
+  }
+
+  // create a mock grpc server that uses delegate to handle rpc calls
+  private void createMockService(
+      BlockMasterWorkerServiceGrpc.BlockMasterWorkerServiceImplBase delegate) throws IOException {
+
+    // make sure to clean up resources
+    cleanUp();
+
+    // create mock service handler
+    BlockMasterWorkerServiceGrpc.BlockMasterWorkerServiceImplBase mockService =
+        mock(
```

Comment:
For the same reason mentioned in the other PR, is it possible to replace this mock with a real child class?

Reply:
Mock is redundant here, we can just use the anonymous `BlockMasterWorkerServiceImplBase` directly. I'm not sure if there would be benefit of creating an explicit child class, since different test methods overrides different methods and there are no obvious common functionalities.

Reply:
Yes we totally can. Your current approach is good.